From: Phillip Wood <phillip.wood@dunelm.org.uk>
Date: Sat, 11 Apr 2020 16:08:18 +0100
Subject: input-settings: fix device list iteration

Dereference the loop variable rather than the original list head. This
fixes a regression introduced in 4413b86a3 ("backends: Replace
ClutterDeviceManager usage in favor of ClutterSeat", 2019-10-04) which
broke button scrolling with trackballs.

(cherry picked from commit 3e967d731a730cb4afb46f115570bb25f54c4195)

Bug: https://gitlab.gnome.org/GNOME/mutter/-/issues/1120
Origin: upstream, 3.36.2
---
 src/backends/meta-input-settings.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/backends/meta-input-settings.c b/src/backends/meta-input-settings.c
index 2028570..5b2cee2 100644
--- a/src/backends/meta-input-settings.c
+++ b/src/backends/meta-input-settings.c
@@ -818,7 +818,7 @@ update_trackball_scroll_button (MetaInputSettings  *input_settings,
 
       for (l = devices; l; l = l->next)
         {
-          device = devices->data;
+          device = l->data;
 
           if (input_settings_class->is_trackball_device (input_settings, device))
             input_settings_class->set_scroll_button (input_settings, device, button);
